### PR TITLE
feat:[ARLS] Update HeciMeExt Library to accomodate all ARL SKUs

### PR DESCRIPTION
**Pull request by sindhuragrandhi**
- Update Common MeChipsetLib to account for ARLS Me Bus. 
   Now, the bus number comes from Platform code: if ARLS bus = 0x80, else bus = 0x0
- Update Heci Pci read calls in HeciMeExtLib to account for both ARLS and ARL U/H  BDF differences.
- Delete ARL specific MeChipset header files as it now uses common header files from Common Package.

### CODE REVIEW

File: Silicon/ArrowlakePkg/Include/Library/HeciMeExtLib.h [hunk 1]
```diff
@@ -12,11 +12,50 @@
...
+/**
+  Reads a 32-bit value from the HECI PCI configuration space.
+
+  @param[in]  Devicefunc  The function number of the HECI device.
+  @param[in]  Regoffset   The offset of the register to read.
+
+  @return The 32-bit value read from the HECI PCI configuration space.
+**/
+UINT32
+HeciPciRead32 (
```
Comment (gdong1) [inline Silicon/ArrowlakePkg/Include/Library/HeciMeExtLib.h:24]: Add EFIAPI for library APIs in both header and C files.

File: Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c [hunk 2]
```diff
@@ -13,12 +13,93 @@
...
+  UINT32 MeBaseAddress;
+  UINT8  Mebus;
+  UINT8  Medevice;
+
+  MeBaseAddress = GetDeviceAddr (PlatformDeviceMe, 0);
+  Mebus = (MeBaseAddress >> 16) & 0xFF;
+  Medevice = (MeBaseAddress >> 8) & 0xFF;
+  MeBaseAddress = PciRead32 (PCI_LIB_ADDRESS(Mebus, Medevice, Function, Regoffset));
+
+  return (MeBaseAddress);
```
Comment (gdong1) [inline Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:47]: () is not required for MeBaseAddress.

File: Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c [hunk 3]
```diff
@@ -13,12 +13,93 @@
...
+  UINTN MeBaseAddress;
+  UINT8 Mebus;
+  UINT8 Medevice;
+
+  MeBaseAddress = GetDeviceAddr (PlatformDeviceMe, 0);
+  Mebus = (MeBaseAddress >> 16) & 0xFF;
+  Medevice = (MeBaseAddress >> 8) & 0xFF;
+  MeBaseAddress = (UINT16)PciRead16 (PCI_LIB_ADDRESS(Mebus, Medevice, Function, Regoffset));
+
+  return (UINT16)(MeBaseAddress);
```
Comment (gdong1) [inline Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:73]: Instead give a PCI value to "MeBaseAddress" and cast,  Maybe combine line 71 and 73 to "return PciRead16 (PCI_LIB_ADDRESS(Mebus, Medevice, Function, Regoffset));"

Comment (sindhuragrandhi) [reply Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:75]: Okay, I can do that. But I felt it becomes too big of a return statement and so I broke it down into multiple lines.

File: Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c [hunk 4]
```diff
@@ -13,12 +13,93 @@
...
+{
+  UINTN MeBaseAddress;
+  UINT8 Mebus;
+  UINT8 Medevice;
+
+  MeBaseAddress = GetDeviceAddr (PlatformDeviceMe, 0);
+  Mebus         = (MeBaseAddress >> 16) & 0xFF;
+  Medevice      = (MeBaseAddress >> 8) & 0xFF;
+
+  return (UINT16)PciRead16 (PCI_LIB_ADDRESS(Mebus, Medevice, Function, Regoffset));
```
Comment (gdong1) [inline Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:73]: cast "(UINT16)" is not required since PciRead16 returns UINT16.

Comment (sindhuragrandhi) [reply Silicon/ArrowlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:73]: ah right. will remove

**Review by gdong1 — APPROVED** [review]

**Pull request merged**